src/clients/uploads-api-client.ts

```diff
@@ -0,0 +1,50 @@
+import globalAxios, { AxiosError, AxiosInstance } from 'axios'
+
+import {
+  Configuration,
+  CreateUploadDestinationResponse,
+  ModelError,
+  UploadsApi,
+  UploadsApiCreateUploadDestinationForResourceRequest,
+} from '../api-models/uploads-api-model'
+import { BASE_PATH } from '../api-models/uploads-api-model/base'
+import { apiErrorFactory } from '../helpers'
+import { UploadsApiClientConfigurationParameters } from '../types/uploads-api-client/uploads-api-client-configuration-parameters'
+import { UploadsApiClientCreateUploadDestinationForResourceRequest } from '../types/uploads-api-client/uploads-api-client-create-upload-destination-for-resource-request'
+
+export class UploadsApiClient {
```

Comment:
Repost of my comment, that was lost for some reason:

---

This wrapping solution seems very heavy handed and would require us to rebuild most of the functionality.

I think a lighter approach should be taken here, and just utilize and extend what we have.

Aproach 1 (sub class):

```ts
interface MyParameters {
  accessToken: string
}

class MyUploadsApi extends UploadsApi {
  constructor(parameters: MyParameters) {
    super(
      {
        isJsonMime: () => true, // stub fake
        accessToken: parameters.accessToken,
      },
      '/',
      axios.create(),
    )
  }
}
```

Approach 2 (extender):

```ts
function extender<C extends Class<any>>(Klass: C) {
  const ax = axios.create()

  return class extends Klass {
    constructor(private parameters: MyParameters) {
      super(
        {
          isJsonMime: () => true, // stub fake
          accessToken: parameters.accessToken,
        },
        '/...',
        ax,
      )
    }
  }
}

class MyUploadsApi extends extender(UploadsApi) {}
```

Both of these examples are incomplete, broken and need more work. But it's just a direction.

Reply:
@moltar : Thanks for your suggestions. I refactored a lighter wrapper. 
I think it may be generated automatically and still keep API error handler after requests.

Reply:
@moltar: I used a [APIClientHelpers ](https://github.com/ScaleLeap/selling-partner-api-sdk/pull/9/files#diff-f5d91263f8f099da5551d6be594ad7ec596961ba3d167f68e656e1272ff887d7R9) to reuse constructor logic.